full-service/src/db/txo.rs

```diff
@@ -1799,14 +1799,21 @@ impl TxoModel for Txo {
                     .on(transaction_logs::id.eq(transaction_input_txos::transaction_log_id)),
             );
 
-        query = query
-            .filter(transaction_logs::id.is_null())
-            .or_filter(transaction_logs::failed.eq(true))
-            .or_filter(
-                transaction_logs::id
+        let inflight_txo_ids = transaction_logs::table
+            .into_boxed()
+            .inner_join(transaction_input_txos::table)
+            .filter(
+                transaction_logs::submitted_block_index
```

Comment:
currently, the build_transaction method is actually reserving its inputs because it gets logged when full-service does the signing, and only build_unsigned_transaction receives no logs.  I believe that this will change some client's to have their build_transaction no longer reserve the input txos as it is making sure that the transaction is submitted when previously that check did not happen.

Do we want to make this change as part of this PR?

Reply:
Do we know what the intent of this was
```rust
            .or_filter(
                transaction_logs::id
                    .is_not_null()
                    .and(transaction_logs::submitted_block_index.is_null()),
```
That seems like it's looking for transactions logs with a non null id, but not submitted.

Reply:
Maybe you're saying the same thing and suggesting further improvements.

I say we keep this PR focused on the current bug at hand. If we want to further improve we do in a follow on pr, even if we want it in this release.

Especially for bug fixes I think it's best to have the PR only contain the bug fix so it's easier to share, patch or revert as necessary

Reply:
agreed. was also trying to not change the semantics here, and think that I got lost in the cascading logic and thought we'd reversed things.